packages/wonder-blocks-modal/components/two-column-modal.js

```diff
@@ -79,31 +126,63 @@ const styles = StyleSheet.create({
     },
 
     column: {
-        boxSizing: "border-box",
-        flex: "0 0 50%",
-
-        // TODO(mdr): Some call sites probably won't want built-in padding. How
-        //     should we handle that possibility?
         paddingTop: 64,
         paddingBottom: 64,
+        flex: "1 1 50%",
+        boxSizing: "border-box",
+
+        // For sidebar
+        minHeight: 464,
     },
 
-    leftColumn: {
+    bgSidebar: {
         background: Color.darkBlue,
         color: Color.white,
+    },
 
-        // TODO(mdr): Some call sites probably won't want built-in padding. How
-        //     should we handle that possibility?
-        paddingLeft: 64,
-        paddingRight: 52,
+    bgSidebarSmall: {
+        flex: "0",
     },
 
-    rightColumn: {
+    bgContents: {
         background: Color.white,
+        position: "relative",
+        flexDirection: "column",
+
+        // For sidebar
+        display: "flex",
+    },
+
+    sidebar: {
+        marginRight: 32,
+    },
+
+    contents: {
+        overflow: "auto",
+        flex: "1 0 0",
+
+        // For sidebar
+        marginLeft: 64,
+        maxWidth: 472,
+    },
+
+    contentsHasFooter: {
+        paddingBottom: 48,
+    },
 
-        // TODO(mdr): Some call sites probably won't want built-in padding. How
-        //     should we handle that possibility?
-        paddingLeft: 52,
-        paddingRight: 64,
+    contentsSmall: {
+        // Center the contents when the sidebar is small
+        margin: "0 auto",
+
+        // Add in some extra padding to the contents to
+        // simulate the extra gutters used for spacing
+        paddingLeft: 64,
+        paddingRight: 32,
+    },
+
+    footer: {
+        position: "absolute",
```

Comment:
Hmm gotcha, I think I see why depending on pure Flexbox is trickier in the two-column case than the one-column case: the grid component has padding, but we want the footer to escape that padding.

I'm still surprised that absolute positioning doesn't disrupt the Flex children here, but I guess, if it works, it works! Long-term, I wonder if this is a hint that we'll need a built-in escape hatch to allow certain grid children to be full-bleed, but I'm okay with this for the modal case if it's working consistently!

Reply:
Yeah, we're definitely going to want an "escape hatch"/"full-bleed" component. For now I just have the footer that I've hacked together but I'm not sure if it makes sense to use absolute positioning for this stuff all the time? I can think about this more and maybe I can come up with a better solution in a follow-up commit.